### PR TITLE
Adding sync run

### PR DESCRIPTION
**Pull request by Reichenbachian**
Adding our manual implementation of a sync run.

### CODE REVIEW

File: src/structify/resources/structure.py [hunk 1]
```diff
@@ -259,6 +259,45 @@ def run_async(
...
+        *,
+        dataset_name: str,
+        text: structure_run_async_params.Variant0Text,
+        custom_instruction: Optional[str] | NotGiven = NOT_GIVEN,
+        # Use the following arguments if you need to pass additional parameters to the API that aren't available via kwargs.
+        # The extra values given here take precedence over values defined on the client or passed to this method.
+        extra_headers: Headers | None = None,
+        extra_query: Query | None = None,
+        extra_body: Body | None = None,
+        timeout: float | httpx.Timeout | None | NotGiven = NOT_GIVEN,
```
Comment (falafel2222) [inline src/structify/resources/structure.py]: Do we need these? They're sorta awkward as part of the API

File: src/structify/resources/structure.py [hunk 2]
```diff
@@ -259,6 +259,45 @@ def run_async(
             ),
             cast_to=object,
         )
+    
+    def run(
```
Comment (falafel2222) [inline src/structify/resources/structure.py:263]: We should probably have a docstring for this since it's public-facing

**Review by falafel2222 — APPROVED** [review]
Yay!

File: src/structify/resources/structure.py (hunk 2 above)
Comment (Reichenbachian) [reply src/structify/resources/structure.py:263]: Ack. Adding.

File: src/structify/resources/structure.py (hunk 1 above)
Comment (Reichenbachian) [reply src/structify/resources/structure.py]: Removed with workaround.

File: src/structify/resources/structure.py [hunk 3]
```diff
@@ -259,6 +259,23 @@ def run_async(
...
+        self,
+        *args,
+        **kwargs,
+    ) -> object:
+        # This is a manually added function. Its goal is to simulate a synchronous run of the async function.
+        # by calling it and then waiting
+        token = self.run_async(*args, **kwargs)
+
+        while True:
```
Comment (falafel2222) [inline src/structify/resources/structure.py:274]: Maybe take in a timeout here, and if we hit that timeout either try to cancel the job or just return?

Comment (Reichenbachian) [reply src/structify/resources/structure.py:274]: Yup. Adding.

**Pull request merged**